linter/rules/yesno_field_label_should_not_contain_yesno.py

```diff
@@ -0,0 +1,13 @@
+from linter.rule import Rule
+from typing import Any, Tuple
+from re import search, IGNORECASE
+
+
+class YesnoFieldLabelShouldNotContainYesno(Rule):
+    def applies_to() -> Tuple[str, ...]:
+        return ('dimension', 'measure')
+
+    def run(self, field: Any) -> bool:
+        if field.get('type') == 'yesno':
+            return not search(r'(yes[_/]no|yesno|yes[_ ]no|yes[_/]No)', field.get('label'), IGNORECASE)
```

Comment:
I may be missing something but doesn't this cut it?

```
return not search(r'(yes[_/ ]*no)', field.get('label'), IGNORECASE)
```

This will match "yes" followed by 0 or more of any of underscore, slash, or space, followed by "no", case-insensitive.